src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs

```diff
@@ -66,7 +68,7 @@ public IReadOnlyList<string> Restore
         {
             get
             {
-                return _restore.ToList();
+                return _restore.Keys.OrderBy(r => r, PathUtility.GetStringComparerBasedOnOS()).ToList();
```

Comment:
It looks this ordering matters some places:
1st [here](https://github.com/NuGet/NuGet.Client/blob/3501ddedc274ac10d4b135856b7593a6bb8a72f1/src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/NoOpRestoreUtilities.cs#L196), 2nd [here](https://github.com/NuGet/NuGet.Client/blob/3501ddedc274ac10d4b135856b7593a6bb8a72f1/src/NuGet.Core/NuGet.ProjectModel/ProjectLockFile/PackagesLockFileUtilities.cs#L78). So better to keep it, as you mentioned it's not bottleneck.

Reply:
#2 is always going to have only 1 project. 
#1 is valid, that affects no-op. 

We can leavel for it. 

PS. When it's not asking for a code change, please allow the reviewer to resolve the thread. It took me some time to find the comment when I clicked the link since it was collapsed.